From 2ac5c1ce326db51cddeae541bb3b62e5850d5e93 Mon Sep 17 00:00:00 2001
From: Davide Tuccilli <d.tuccilli@gmail.com>
Date: Thu, 19 Nov 2015 04:04:29 +0100
Subject: [PATCH] Added some geometry stuff, shadow maps, librocket integration

---
 frustum.cpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100644 frustum.cpp

diff --git a/frustum.cpp b/frustum.cpp
new file mode 100644
index 0000000..338c1e4
--- /dev/null
+++ b/frustum.cpp
@@ -0,0 +1,21 @@
+#include <fuse/geometry/frustum.hpp>
+
+using namespace fuse;
+
+FUSE_DEFINE_ALIGNED_ALLOCATOR_NEW(frustum, 16)
+
+frustum::frustum(const XMMATRIX & viewProjection)
+{
+
+	XMMATRIX viewProjectionT = XMMatrixTranspose(viewProjection);
+
+	m_planes[FUSE_FRUSTUM_PLANE_NEAR].set_plane_vector(viewProjectionT.r[2]);
+	m_planes[FUSE_FRUSTUM_PLANE_FAR].set_plane_vector(viewProjectionT.r[3] - viewProjectionT.r[2]);
+
+	m_planes[FUSE_FRUSTUM_PLANE_TOP].set_plane_vector(viewProjectionT.r[3] - viewProjectionT.r[1]);
+	m_planes[FUSE_FRUSTUM_PLANE_BOTTOM].set_plane_vector(viewProjectionT.r[3] + viewProjectionT.r[1]);
+
+	m_planes[FUSE_FRUSTUM_PLANE_LEFT].set_plane_vector(viewProjectionT.r[3] + viewProjectionT.r[0]);
+	m_planes[FUSE_FRUSTUM_PLANE_RIGHT].set_plane_vector(viewProjectionT.r[3] - viewProjectionT.r[0]);
+
+}
\ No newline at end of file
-- 
1.9.5.msysgit.0

